youtube_dl/extractor/teleportal.py

```diff
@@ -0,0 +1,40 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class TeleportalIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?teleportal\.ua(/ua)?/(?P<id>[0-9a-z-/]+)'
+    _TEST = {
+        'url': 'https://teleportal.ua/ua/show/stb/master-cheff/bitva-sezonov/vypusk-3',
+        'md5': '07bd056c45b515fa9cc0202b8403df41',
+        'info_dict': {
+            'id': 'show/stb/master-cheff/bitva-sezonov/vypusk-3',
+            'ext': 'mp4',
+            'title': 'МастерШеф. Битва сезонів 3 випуск: найогидніший випуск сезону!',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'description': r're:^<p>Не пропустіть.*',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        backend_url = 'https://tp-back.starlight.digital/ua/{}'.format(video_id)
+        metadata = self._download_json(backend_url, video_id)
+        api_metadata = self._download_json('https://vcms-api2.starlight.digital/player-api/{}?referer=https://teleportal.ua/&lang=ua'.format(metadata['hash']), video_id)
+
+        try:
+            thumbnail = api_metadata['video'][0]['poster']
+        except (KeyError, IndexError):
+            thumbnail = None
+
+        return {
+            'id': video_id,
+            'title': metadata.get('title'),
+            'description': metadata.get('description'),
+            'real_id': metadata.get('id'),
+            'hash': metadata.get('hash'),
+            'thumbnail': thumbnail,
+            'formats': self._extract_m3u8_formats(api_metadata['video'][0]['mediaHls'], video_id, 'mp4'),
```

Comment:
Better to do this first, at your line 26 (I've added comments to explain that wouldn't be in the final code):
```
        title = metadata['title']  # mandatory field
        fmt_url = try_get(api_metadata, lambda x: x['video'][0]['mediaHls'])
        formats = []
        if fmt_url:
            formats.extend(self._extract_m3u8_formats(fmt_url, video_id, 'mp4')
        self._check_formats(formats)  #  include this line if the URLs in the m3u8 might sometimes fail (403, 404, eg)
        self._sort_formats(formats)  #  will raise exception if no formats
```